praxis-isncsci/ui
# Use template in app markup

Closes #86 

## Problem

When building the test app we are using duplicate markup which exists in `src/web/praxisIsncsciAppLayout/appLayoutTemplate.ts`.

We should use the template markup straight into index.html. This should be automated during the build process.

## Solution

We are adding a `postbuild` step that will replace the constant `{{app-template}}` in the `build/index.html` with the template in `src/web/praxisIsncsciAppLayout/appLayoutTemplate.ts`. For the template to work in our build pipeline, we had to add a config to `Rollup` so we can export the template function as `cjs` and use it in **node**.

## Testing

- [x] 1. Can inject the template markup into `index.html`

<details>
  <summary>Test case</summary>

  1. Navigate to `assets/index.html`
  2. Confirm the string `{{app-template}}` is present
  3. Execute `npm run build`
  4. Open `build/index.html`
  5. Confirm the string `{{app-template}}` was replaced with the template markup

</details>
